src/app/operations/intraledger_payment.rs

```diff
@@ -0,0 +1,64 @@
+use anyhow::Result;
+
+use rust_decimal::Decimal;
+
+use crate::{app::App, client::types::Wallet, errors::payment_error::PaymentError};
+
+impl App {
+    pub async fn intraledger_payment(
+        &self,
+        username: String,
+        wallet: Wallet,
+        cents: Option<Decimal>,
+        sats: Option<Decimal>,
+        memo: Option<String>,
+    ) -> Result<()> {
+        if wallet == Wallet::Btc && sats.is_none() {
+            return Err(PaymentError::AmountNotSpecifiedBTC.into());
+        }
+
+        if wallet == Wallet::Usd && cents.is_none() {
+            return Err(PaymentError::AmountNotSpecifiedUSD.into());
+        }
+
+        match wallet {
```

Comment:
I would add the amounts here and have a match with multiple cases. Then you don't need the early returns and you don't need the `expect` calls later on:

```
match (wallet, sats, cents) {
  (Wallet::Btc, None, _ ) => return Err(anyhow!("Amount not specified in Btc"))
  (Wallet::Btc, Some(sats), _) => <payment code>
  etc.
}
```